pkg/custom-plugins/spec_component.go

```diff
@@ -0,0 +1,211 @@
+package customplugins
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	apiv1 "github.com/leptonai/gpud/api/v1"
+	"github.com/leptonai/gpud/components"
+	"github.com/leptonai/gpud/pkg/log"
+)
+
+// NewInitFunc creates a new component initializer for the given plugin spec.
+func (spec *Spec) NewInitFunc() components.InitFunc {
+	if spec == nil {
+		return nil
+	}
+	return func(gpudInstance *components.GPUdInstance) (components.Component, error) {
+		cctx, ccancel := context.WithCancel(gpudInstance.RootCtx)
+		c := &component{
+			ctx:    cctx,
+			cancel: ccancel,
+			spec:   spec,
+		}
+		return c, nil
+	}
+}
+
+var _ components.Component = &component{}
+
+type component struct {
```

Comment:
shall we rename the filename to component.go as others?

Reply:
done